src/utils/isRandomChance.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * Does a dice roll
+ * @param {float} chance float between 0-1 to compare dice roll against
+ * @returns {bool} true if the dice roll was equal to or lower than the given chance, false otherwise
+ */
+export default function isRandomChance(chance) {
```

Comment:
This function name a is a bit confusing to me, as it doesn't quite map to what this function is computing. I read `isRandomChance` as "is this thing a random chance?" but it's determining if a given value is greater than a random value. Maybe something like `isRandomNormalValueLessThan` or `isRandomlyLessThan` instead?

Reply:
i know what you mean, and i'm open to changing it, but i am going to attempt to defend the name first. i actually started using this type of method in a random project recently after dealing with mocking `Math.random` in a bunch of tests, very similar to the resource factory stuff. i started to think that not only is mocking `Math.random` a pain, but the lines of code often don't make all that much sense either. for example, i was using the name `diceRoll` as a var to try to help document what was going on in code like:

```javascript
const diceRoll = Math.random()
if (diceRoll < .5) {
    doSomething()
}
```

so, the name `isRandomChance` seemed to nicely solve both of these problems for me. it's easy to mock to return `true` or `false` and it makes the above code much easier to write, and to read.

```javascript
if (isRandomChance(.5)) {
    doSomething()
}
```

so, i sort of purposely avoided exposing the under the hood concept in favor of the way the method reads when used. all that said, i really am open to changing it. i just wanted to explain the whole story i had in mind.

Reply:
I understand what you're saying. I think this utility function makes sense and fits well in the codebase, but the name just doesn't map to what it's doing for me. 😅

I think a much more explicit name like `isRandomNumberLessThan` will yield the same functional benefits while also making the code easier to understand. With `isRandomChance`, I feel like I'll have to keep looking at the code's implementation to know what it does.

Reply:
aye! `I'll have to keep looking at the code's implementation to know what it does.` i don't want that 😄 . so i've updated to `isRandomNumberLessThan` now.

Reply:
Awesome, thank for being flexible with the name! The code reads much more easily to me now. 😄